### PR TITLE
Switch theme prop

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Add theme prop to switch to enable light and dark mode

## Related issue
Working towards #1214 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

File: packages/react/src/component-tests/IcSwitch/IcSwitchTestData.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React, { useState, useRef } from "react";
-import { IcButton, IcSwitch } from "../../components";
+import { IcButton, IcSwitch, IcTypography } from "../../components";
```
Comment (ad9242) [inline packages/react/src/component-tests/IcSwitch/IcSwitchTestData.tsx:2]:
```suggestion
import { IcButton, IcSwitch } from "../../components";
```
IcTypography doesn't appear to be used

File: packages/web-components/src/global/color-mode.css [hunk 2]
```diff
@@ -701,6 +763,38 @@ ic-theme,
   );
   --ic-stepper-compact-step-completed-text: var(--ic-status-success-contrast);
 
+  /* Switch */
+  --ic-switch-checked-bg: var(--ic-color-background-active);
```
Comment (ad9242) [inline packages/web-components/src/global/color-mode.css:767]:
```suggestion
  --ic-switch-checked-bg: var(--ic-color-background-active-dark);
```

File: packages/web-components/src/global/color-mode.css [hunk 3]
```diff
@@ -461,6 +491,38 @@ ic-theme,
     );
     --ic-stepper-compact-step-completed-text: var(--ic-status-success-contrast);
 
+    /* Switch */
+    --ic-switch-checked-bg: var(--ic-color-background-active);
```
Comment (ad9242) [inline packages/web-components/src/global/color-mode.css:495]:
```suggestion
    --ic-switch-checked-bg: var(--ic-color-background-active-dark);
```

**Review by ad9242 — CHANGES_REQUESTED** [review]
found a bit of an issue in the playground. if you have the switch checked & then change theme, the checked color is not updating. I think it might be something to do with using the `--ic-color-background-active` token, which won't be updating at the component scope. I think best to use the `--ic-color-background-active-dark` token

**Review by ad9242 — APPROVED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Pull request merged**